peps/pep-0748.rst

```diff
@@ -0,0 +1,735 @@
+  ``get_annotations()`` when the format is ``FORWARDREF``. The existing class
+  :py:class:`typing.ForwardRef` will become an alias of this class. Its members include:
+
+  * ``__forward_arg__``: the string argument of the forward reference
+  * ``evaluate(globals=None, locals=None, type_params=None, owner=None)``: a method that attempts to evaluate
+    the forward reference. The ``ForwardRef`` object may hold a reference to the
+    globals and other namespaces of the object that it originated from. If so, these
+    namespaces may be used to evaluate the forward reference. The *owner* argument
+    may be the object that holds the original annotation, such as the class or module
+    object; it is used to extract the globals and locals namespaces if these are not
+    provided.
+  * ``_evaluate()``, with the same interface as the existing ``ForwardRef._evaluate``
+    method. It will be undocumented and immediately deprecated. It is provided for
+    compatibility with existing users of ``typing.ForwardRef``.
+
+* ``call_annotate_function(func: Callable, format: Format)``: a helper for calling
+  an ``__annotate__`` function with a given format. If the function does not support
+  this format, ``call_annotate_function()`` will set up a "fake globals" environment,
+  as described in :pep:`649`, and use that environment to return the desired annotations
+  format.
+* ``call_evaluate_function(func: Callable | None, format: Format)``: similar to
+  ``call_annotate_function``, but does not rely on the function returning an annotations
+  dictionary. This is intended to be used for evaluating deferred attributes introduced by
+  :pep:`695` and :pep:`696`; see below for details. *func* may be ``None``
+  for convenience; if ``None`` is passed, the function also returns ``None``.
+
+A new function is also added to the :py:mod:`!typing` module, ``typing.evaluate_forward_ref``.
+This function is a wrapper around the ``ForwardRef.evaluate`` method, but it performs
+additional work that is specific to type hints. For example, it recurses into complex
+types and evaluates additional forward references within these types.
+
+Open issues
+-----------
+
+What should this module be called? Some ideas:
+
+- ``annotations``: The most obvious name, but it may cause confusion with the existing
+  ``from __future__ import annotations``. There is a PyPI package :pypi:`annotations`,
+  but it had only a single release in 2015 and looks abandoned.
+- ``annotools``: Analogous to :py:mod:`itertools` and :py:mod:`functools`, but "anno" is a less
+  obvious abbreviation than "iter" or "func". As of this writing, there
+  is no PyPI package with this name.
+- ``annotationtools``: A more explicit version. There is a PyPI package
+  :pypi:`annotationtools`, which had a release in 2023.
+
+Rejected alternatives
+---------------------
+
+*Add the functionality to the :py:mod:`inspect` module*: As described above, the
```

Comment:
This (and the later two) aren't linking to docs.python.org

Reply:
Thanks. I think I'll just drop the links; there might be some magical incantation to get links between `*` but it's probably not worth it.